src/client/src/Stream.c

```diff
@@ -363,9 +363,16 @@ STATUS freeStream(PKinesisVideoStream pKinesisVideoStream)
     freeMetadataTracker(&pKinesisVideoStream->eosTracker);
     freeMetadataTracker(&pKinesisVideoStream->metadataTracker);
 
+    // unlock the stream freeFrameOrderCoordinator acquires the FrameOrderCoordinator mutex, we cannot acquire this
+    // mutex while holding the stream mutex or else we will deadlock with putFrame which first acquires
+    // FrameOrderCoordinator mutex and then the stream mutex
+    pKinesisVideoClient->clientCallbacks.unlockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoStream->base.lock);
+
     // Free FrameOrderCoordinator
     freeFrameOrderCoordinator(pKinesisVideoStream, &pKinesisVideoStream->pFrameOrderCoordinator);
 
+    pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoClient->clientCallbacks.customData, pKinesisVideoStream->base.lock);
+
```

Comment:
We acquire the lock again in the next line. Is that because previously some function on the path unlocked the mutex already?

Reply:
The assumption in this function is that we have the lock LN 406-407 we unlock and free this lock.